HttpClientTimeoutTests can occasionally fail due to unreachable network

**Jetty version**
9.4.x

**Java version**
11.0.7+

**OS type/version**
Linux

**Description**
The tests that utilize the IP 10.255.255.1 can sometimes result in an unexpected `java.net.SocketException: Network is unreachable (connect failed)` when the test is attempting to test for connect timeouts.

```
org.eclipse.jetty.http.client.HttpClientTimeoutTest.testConnectTimeoutIsCancelledByShorterRequestTimeout(Transport)[3]  Time elapsed: 0.006 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 
	at org.eclipse.jetty.http.client.HttpClientTimeoutTest.assumeConnectTimeout(HttpClientTimeoutTest.java:523)
	at org.eclipse.jetty.http.client.HttpClientTimeoutTest.testConnectTimeoutIsCancelledByShorterRequestTimeout(HttpClientTimeoutTest.java:345)
Caused by: java.net.SocketException: Network is unreachable (connect failed)
	at org.eclipse.jetty.http.client.HttpClientTimeoutTest.assumeConnectTimeout(HttpClientTimeoutTest.java:511)
	at org.eclipse.jetty.http.client.HttpClientTimeoutTest.testConnectTimeoutIsCancelledByShorterRequestTimeout(HttpClientTimeoutTest.java:345)
```

The method `HttpClientTimeoutTest.assumeConnectTimeout()` should be updated to handle this situation.

https://github.com/eclipse/jetty.project/blob/65de149f84b68e23fb2a2a849e80b06850c8cd29/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientTimeoutTest.java#L503-L525

--- a/pom.xml
+++ b/pom.xml
@@ -44,7 +44,7 @@
     <jpms-module-name>${bundle-symbolic-name}</jpms-module-name>
 
     <!-- some maven plugins versions -->
-    <maven.surefire.version>3.0.0-M4</maven.surefire.version>
+    <maven.surefire.version>3.0.0-M5</maven.surefire.version>
     <maven.compiler.plugin.version>3.8.1</maven.compiler.plugin.version>
     <maven.dependency.plugin.version>3.1.1</maven.dependency.plugin.version>
     <maven.resources.plugin.version>3.1.0</maven.resources.plugin.version>

--- a/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientTimeoutTest.java
+++ b/tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientTimeoutTest.java
@@ -57,6 +57,7 @@ import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.junit.jupiter.api.Assumptions;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.opentest4j.TestAbortedException;
 
 import static org.eclipse.jetty.http.client.Transport.UNIX_SOCKET;
 import static org.junit.jupiter.api.Assertions.assertArrayEquals;
@@ -509,7 +510,7 @@ public class HttpClientTimeoutTest extends AbstractTest<TransportScenario>
             // connect to them will hang the connection attempt, which is
             // what we want to simulate in this test.
             socket.connect(new InetSocketAddress(host, port), connectTimeout);
-            // Abort the test if we can connect.
+            // Fail the test if we can connect.
             fail("Error: Should not have been able to connect to " + host + ":" + port);
         }
         catch (SocketTimeoutException x)
@@ -520,7 +521,7 @@ public class HttpClientTimeoutTest extends AbstractTest<TransportScenario>
         catch (Throwable x)
         {
             // Abort if any other exception happens.
-            fail(x);
+            throw new TestAbortedException("Not able to validate connect timeout conditions", x);
         }
     }
 

